src/main/java/org/example/Phone.java

```diff
@@ -0,0 +1,112 @@
+package org.example;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public abstract class Phone {
+    /** Abstract Class for a Phone object that emulates the behavior of a real phone.
+     * This is the parent class for the Iphone and Samsung child classes.
+     * You can add contacts and fetch them, call them and send messages.
+     * You can display the call history, retrieve first or second message for each contact.
+     * Each call or message drains the battery and you can use getBatteryLife() to check current levels.
+     */
+    protected final int batteryLife;
+    protected String color;
+    protected String material;
+    protected String imei;
+    protected int currentBatteryLife;
+    LinkedHashMap<String, Contact> contactList = new LinkedHashMap<>();
+    private ArrayList<String> callHistory = new ArrayList<>();
+
+    protected Phone(int batteryLife, String color, String material, String imei) {
+        this.batteryLife = batteryLife;
+        this.color = color;
+        this.material = material;
+        this.imei = imei;
+        currentBatteryLife = batteryLife;
+    }
+
+    public void addContact(int id, String phoneNumber, String firstName, String lastName) {
+        Contact contact = new Contact(id, phoneNumber, firstName, lastName);
```

Comment:
this method should receive Contact object as a parameter instead of all its fields.